chain/chain/src/test_utils.rs

```diff
@@ -1170,16 +1171,21 @@ mod test {
         let create_receipt_from_receiver_id =
             |receiver_id| Receipt::new_balance_refund(&receiver_id, 0);
         let mut rng = rand::thread_rng();
-        let receipts = (0..100)
+        let receipts = (0..3000)
             .map(|_| {
                 let random_number = rng.gen_range(0, 1000);
                 create_receipt_from_receiver_id(format!("test{}", random_number))
             })
             .collect::<Vec<_>>();
-        assert_eq!(
-            runtime_adapter.naive_build_receipt_hashes(&receipts),
-            runtime_adapter.build_receipts_hashes(&receipts)
-        );
+        let start = Instant::now();
+        let naive_result = runtime_adapter.naive_build_receipt_hashes(&receipts);
+        let naive_duration = start.elapsed();
+        let start = Instant::now();
+        let prod_result = runtime_adapter.build_receipts_hashes(&receipts);
+        let prod_duration = start.elapsed();
+        assert_eq!(naive_result, prod_result);
+        // production implementation is at least 50% faster
```

Comment:
I wonder whether a proper benchmark (using criterion for example) would be better. Are you confident that such assertion would hold on every machine?

Reply:
I tried it many times on an Azure VM and an old laptop of mine (quite a slow machine) and all passed. It also passed in CI here (though that is only 1 run, which is not enough to say that will probably always pass). The condition it's checking for is pretty weak, especially when there are more shards. I was observing a 3x-4x speedup for 10 shards for example.

So I am quite confident it will always pass. But we could still do a more precise benchmark if you think there is value in doing so.

Reply:
I see. If you are sure it will pass I think it is fine as is.